index.js

```diff
@@ -294,13 +301,12 @@ module.exports = function (filename, opts) {
     onReady,
 
     onDrain: onLoad(function (fn) {
-      let blockIndexes = Object.keys(blocksToBeWritten)
-      if (blockIndexes.length == 0) fn()
+      if (blocksToBeWritten.size === 0) fn()
       else {
-        const latestBlockIndex = blockIndexes[blockIndexes.length-1]
-        const drains = waitingDrain[latestBlockIndex] || []
+        const latestBlockIndex = last(blocksToBeWritten.keys())
```

Comment:
isn't there a more idiomatic way, this is O(n)?

Reply:
It's O(n) in CPU but O(1) in storage, e.g. it doesn't create an array. I think Object.keys() may do O(n) to create and return an array, so it might be O(n) in both CPU and Storage. I'll make a jsperf microbenchmark to make sure

Reply:
https://jsbench.me/eakievd9nz/1

On Chrome and Safari, Map's `last()` is about 10x faster than `Object.keys()` and `length-1`.

On Firefox, `Object.keys()` and `length-1` is about 1.5x faster than Map's `last()`.

So because Node.js is running V8, probably has similar behavior than Chrome.

Reply:
Ahh I just read the whole thing again. Didn't see we already had the object.keys a bit up. Also we are hopefully never going to have that many values in this map. Still really strange to see those numbers for the different browsers.